Packs/AWS-SQS/Integrations/AWS-SQS/AWS-SQS.py

```diff
@@ -147,33 +147,36 @@ def parse_incident_from_finding(message):
     return incident
 
 
-def fetch_incidents(aws_client, aws_queue_url):
+def fetch_incidents(aws_client, aws_queue_url, max_fetch):
     try:
         client = aws_client.aws_session(service='sqs')
-        messages = client.receive_message(
-            QueueUrl=aws_queue_url,
-            MaxNumberOfMessages=10,
-            VisibilityTimeout=5,
-            WaitTimeSeconds=5,
-        )
-
         receipt_handles = []  # type: list
         incidents = []  # type: list
-
-        if "Messages" not in messages.keys():
-            if demisto.command() == 'fetch-incidents':
-                demisto.incidents([])
-            return messages, incidents, receipt_handles
-
-        for message in messages["Messages"]:
-            receipt_handles.append(message['ReceiptHandle'])
-            incidents.append(parse_incident_from_finding(message))
+        while len(incidents) < max_fetch:
+            messages = client.receive_message(
+                QueueUrl=aws_queue_url,
+                MaxNumberOfMessages=10,
+                VisibilityTimeout=5,
+                WaitTimeSeconds=5,
+            )
+
+            if "Messages" not in messages.keys():
+                if len(incidents) == 0:
+                    if demisto.command() == 'fetch-incidents':
+                        demisto.incidents([])
+                    return messages, incidents, receipt_handles
+                else:
+                    break
```

Comment:
Make sure to review the second commit from this morning. I did catch that scenario you describe where the number of retrievable incidents is less than the configured max_fetch parameter. If the queue ever runs out of messages then the response will not contain the `Messages` key and the break statement will exit the while loop.

Reply:
you're right, it makes max_fetch works as it's expected to